tests/Mutator/DefinitionMutatorTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Nusje2000\ComposerMonolith\Tests\Mutator;
+
+use LogicException;
+use Nusje2000\ComposerMonolith\Composer\DefinitionMutator;
+use PHPStan\Testing\TestCase;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class DefinitionMutatorTest extends TestCase
+{
+    public function testSetDependency(): DefinitionMutator
+    {
+        $mutator = $this->createMutator();
+
+        $mutator->setDependency('some/package-2', '^2.0');
+        $this->addToAssertionCount(1);
+
+        return $mutator;
+    }
+
+    /**
+     * @depends testSetDependency
+     */
+    public function testRemoveDependency(DefinitionMutator $mutator): DefinitionMutator
+    {
+        $mutator->removeDependency('some/package');
+        $this->addToAssertionCount(1);
+
+        return $mutator;
+    }
+
+    /**
+     * @depends testRemoveDependency
+     */
+    public function testSetDevDependency(DefinitionMutator $mutator): DefinitionMutator
+    {
+        $mutator->setDevDependency('some/dev-package-2', '^2.0');
+        $this->addToAssertionCount(1);
+
+        return $mutator;
+    }
+
+    /**
+     * @depends testSetDevDependency
+     */
+    public function testRemoveDevDependency(DefinitionMutator $mutator): DefinitionMutator
+    {
+        $mutator->removeDevDependency('some/dev-package');
+        $this->addToAssertionCount(1);
+
+        return $mutator;
+    }
+
+    /**
+     * @depends testRemoveDevDependency
+     */
+    public function testSetReplace(DefinitionMutator $mutator): DefinitionMutator
+    {
+        $mutator->setReplace('some/replace-2', 'self.version');
+        $this->addToAssertionCount(1);
+
+        return $mutator;
+    }
+
+    /**
+     * @depends testSetReplace
+     */
+    public function testRemoveReplace(DefinitionMutator $mutator): DefinitionMutator
+    {
+        $mutator->removeReplace('some/replace');
+        $this->addToAssertionCount(1);
+
+        return $mutator;
+    }
+
+    /**
+     * @depends testRemoveReplace
+     */
+    public function testSave(DefinitionMutator $mutator): void
+    {
+        $mutator->save();
+
+        self::assertJsonFileEqualsJsonFile(__DIR__ . '/result_composer.json', __DIR__ . '/initial_composer.json');
+    }
+
+    public function testSaveException(): void
+    {
+        $mutator = $this->createMutator();
+        $mutator->removeReplace('some/replace');
+        unlink(__DIR__ . '/initial_composer.json');
+        $this->expectException(LogicException::class);
```

Comment:
I'd expect some library-specific exception instead of general `LogicException`

Reply:
Continue conversation in https://github.com/nusje2000/composer-monolith/pull/11#discussion_r371122958.